### PR TITLE
[APPC-1445] Default card unselected with two decimal place numbers

### PR DESCRIPTION
**Pull request by JoaoRaimundo**
**What does this PR do?**

1-Improve Top Up value comparison to chips value, so 4.5 and 4.50 can match
2-Fix match of payment or Top Up value comparison to minim transaction for referrals to be equal or greater and not only greater.
3-Add build variable for Aptoide's package name.

**Database changed?**

    No

**Where should the reviewer start?**

- [ ] TopUpData.kt
- [ ] TopUpFragment.kt
- [ ] TopUpFragmentPresenter.kt
- [ ] TopUpFragmentView.kt
- [ ] TopUpInteractor.kt
- [ ] BarcodeCaptureActivity.java
- [ ] GamificationInteractor.kt
- [ ] FiatValue.kt
- [ ] InAppPurchaseInteractor.java

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [APPC-1445](https://aptoide.atlassian.net/browse/APPC-1445)

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-1445](https://aptoide.atlassian.net/browseAPPC-1445)

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

### CODE REVIEW

File: app/src/main/java/com/asfoundation/wallet/topup/TopUpFragment.kt [hunk 1]
```diff
@@ -453,31 +453,23 @@ class TopUpFragment : DaggerFragment(), TopUpFragmentView {
   private fun setChipOnClickListener(index: Int) {
     when (index) {
       0 -> {
-        default_chip1.setOnCheckedChangeListener { _, isChecked ->
-          if (isChecked) {
-            chipClickSubject?.onNext(index)
-          }
+        default_chip1.setOnClickListener {
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/topup/TopUpFragment.kt]: There's no need for this bracket. For ex:

    when (index) {
      0 -> default_chip1.setOnClickListener { chipClickSubject?.onNext(index) }
      1 -> default_chip2.setOnClickListener { chipClickSubject?.onNext(index) }
      2 -> default_chip3.setOnClickListener { chipClickSubject?.onNext(index) }
      3 -> default_chip4.setOnClickListener { chipClickSubject?.onNext(index) }

File: app/src/main/java/com/asfoundation/wallet/topup/TopUpFragment.kt [hunk 2]
```diff
@@ -362,7 +362,7 @@ class TopUpFragment : DaggerFragment(), TopUpFragmentView {
     }
   }
 
-  override fun unselectChips() {
+  override fun deselectChips() {
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/topup/TopUpFragment.kt:365]: Nice to read : [unselect vs deselect](https://english.stackexchange.com/a/106721)

**Review by carvalho2707 — CHANGES_REQUESTED** [review]

**Review by carvalho2707 — APPROVED** [review]

File: app/src/main/java/com/asfoundation/wallet/topup/TopUpFragmentPresenter.kt [hunk 3]
```diff
@@ -109,7 +109,10 @@ class TopUpFragmentPresenter(private val view: TopUpFragmentView,
 
   private fun handleManualAmountChange(packageName: String) {
     disposables.add(view.getEditTextChanges()
-        .throttleLatest(700, TimeUnit.MILLISECONDS, viewScheduler, true)
+        .doOnNext {
+          resetValues(it)
+        }
```
Comment (JDSM01) [inline app/src/main/java/com/asfoundation/wallet/topup/TopUpFragmentPresenter.kt]: Line break not needed

File: app/src/main/java/com/asfoundation/wallet/topup/TopUpFragmentPresenter.kt [hunk 4]
```diff
@@ -118,14 +121,20 @@ class TopUpFragmentPresenter(private val view: TopUpFragmentView,
               .map { value ->
                 return@map updateConversionValue(value.amount, topUpData)
```
Comment (JDSM01) [inline app/src/main/java/com/asfoundation/wallet/topup/TopUpFragmentPresenter.kt]: return@map not needed and consequently line break not needed too

File: app/src/main/java/com/asfoundation/wallet/topup/TopUpFragmentPresenter.kt [hunk 5]
```diff
@@ -118,14 +121,20 @@ class TopUpFragmentPresenter(private val view: TopUpFragmentView,
...
               }
-              .doOnError { it.printStackTrace() }
-              .onErrorResumeNext(Observable.empty())
               .observeOn(viewScheduler)
-              .filter { isConvertedValueAvailable(it)}
+              .filter { isConvertedValueAvailable(it) }
               .doOnComplete {
                 view.setConversionValue(topUpData)
```
Comment (JDSM01) [inline app/src/main/java/com/asfoundation/wallet/topup/TopUpFragmentPresenter.kt]: Line break not needed

File: app/src/main/java/com/asfoundation/wallet/topup/TopUpFragmentPresenter.kt [hunk 6]
```diff
@@ -118,14 +121,20 @@ class TopUpFragmentPresenter(private val view: TopUpFragmentView,
...
-                handleInsertedValue(packageName, topUpData)
               }
+              .flatMap {
+                interactor.getLimitTopUpValues()
+                    .toObservable()
+                    .flatMap {
+                      handleInsertedValue(packageName, topUpData, it)
```
Comment (JDSM01) [inline app/src/main/java/com/asfoundation/wallet/topup/TopUpFragmentPresenter.kt]: Line break not needed

**Review by JDSM01 — CHANGES_REQUESTED** [review]
Shouldn't we only enable the next button after getting the bonus? Otherwise the bonus that the user will see in the end will not be the updated one. Or we can recalculate the bonus in a later phase of the topup

**Review by JDSM01 — APPROVED** [review]

**Pull request merged**